### PR TITLE
Make Andersen emit PointsToGraphs with RegisterSetNodes

### PR DESCRIPTION
**Pull request by haved**
Instead of emitting one `PointsToGraph::RegisterNode` per `rvsdg::output*` represented in the PointsToGraph, it groups all `rvsdg::output*`s associated with the same `PointerObject`, and creates one `RegisterSetNode` for them. This should enable the future removal of `RegisterNode`s.

While translating the tests, I observed that the number of `RegisterSetNode`s was about 50-90% of the amount of `RegisterNode`s the test's PointsToGraph used to have, a nice improvement.

### CODE REVIEW

File: jlm/llvm/opt/alias-analyses/Andersen.cpp [hunk 1]
```diff
@@ -430,7 +430,7 @@ Andersen::AnalyzePhi(const phi::node & phi)
 
     // Map the output register to the recursion result's pointer object
     auto & outputRegister = *rv;
-    Set_->MapRegisterToExistingPointerObject(outputRegister, argumentRegisterPO);
+    Set_->MapRegisterToExistingPointerObject(outputRegister, resultRegisterPO);
```
Comment (haved) [inline jlm/llvm/opt/alias-analyses/Andersen.cpp:433]: This modification would not affect the PointsToGraph previously, but now it actually makes a difference if the phi node's output is mapped to the result or the argument. (As you can see, the comment was previously a lie, now the comment is true)

File: tests/jlm/llvm/opt/alias-analyses/TestAndersen.cpp [hunk 2]
```diff
@@ -8,12 +8,12 @@
...
 #include <jlm/llvm/opt/alias-analyses/Andersen.hpp>
-#include <jlm/llvm/opt/alias-analyses/PointerObjectSet.hpp>
 #include <jlm/llvm/opt/alias-analyses/PointsToGraph.hpp>
-#include <jlm/rvsdg/node.hpp>
-#include <jlm/rvsdg/view.hpp>
 #include <jlm/util/Statistics.hpp>
 
+// TODO: Remove
+#include <jlm/rvsdg/view.hpp>
+
```
Comment (phate) [inline tests/jlm/llvm/opt/alias-analyses/TestAndersen.cpp:16]: This should be removed again, I guess?

**Review by phate — APPROVED** [review]
LGTM

File: tests/jlm/llvm/opt/alias-analyses/TestAndersen.cpp (hunk 2 above)
Comment (haved) [reply tests/jlm/llvm/opt/alias-analyses/TestAndersen.cpp:16]: Darn, of course

**Pull request merged**